src/video/wayland/SDL_waylandwindow.c

```diff
@@ -834,8 +832,6 @@ decoration_frame_configure(struct libdecor_frame *frame,
             GetFullScreenDimensions(window, &width, &height, NULL, NULL);
         }
 
-        wind->needs_resize_event = SDL_TRUE;
```

Comment:
`Wayland_HandleResize()` will determine if a resize event is needed based on the new width, height, and scaling factor.